### PR TITLE
Add more tracing information

### PR DESCRIPTION
**Pull request by nambrot**
This PR adds more tracing information to the `tracing` spans. Most information is hidden behind debug to avoid spamming the logs for now. The other useful thing is that a lot of methods that return result will now log the error

### CODE REVIEW

File: rust/abacus-base/src/types/checkpoint_syncer.rs [hunk 1]
```diff
@@ -77,20 +78,25 @@ pub enum CheckpointSyncers {
 
 #[async_trait]
 impl CheckpointSyncer for CheckpointSyncers {
+    #[instrument(err, skip(self), level = "debug")]
```
Comment (tkporter) [inline rust/abacus-base/src/types/checkpoint_syncer.rs:81]: ah doing it at the `CheckpointSyncers` level is much nicer than what I did [here](https://github.com/abacus-network/abacus-monorepo/pull/361/files#diff-b5624ab7cd098615872f3a72f5b36d769ef28e9ee27d6f748b3edffde86376b9R48), i'm gonna remove in favor of this

File: rust/abacus-base/src/types/multisig.rs [hunk 2]
```diff
@@ -104,6 +107,9 @@ impl MultisigCheckpointSyncer {
         if latest_indices.is_empty() {
             return Ok(None);
         }
+
+        debug!(latest_indices=?latest_indices, "Fetched latest indices from checkpoint syncers");
```
Comment (tkporter) [inline rust/abacus-base/src/types/multisig.rs:111]: Wonder if this is worth putting above the `is_empty()` check?

File: rust/chains/abacus-ethereum/src/inbox.rs [hunk 3]
```diff
@@ -233,7 +233,7 @@ where
         Ok(self.contract.remote_domain().call().await?)
     }
 
-    #[tracing::instrument(err)]
+    #[tracing::instrument(err, skip(proof))]
```
Comment (tkporter) [inline rust/chains/abacus-ethereum/src/inbox.rs:236]: should this also skip self?

**Review by tkporter — APPROVED** [review]

File: rust/chains/abacus-ethereum/src/inbox.rs (hunk 3 above)
Comment (nambrot) [reply rust/chains/abacus-ethereum/src/inbox.rs:236]: Yeah so I was thinking about skipping self here, but i have found myself finding it quite useful at times to identify the inbox, signer address, etc. so I was inclined to keep it for now. I do think it should really be kept in a different span ideally

**Pull request merged**